picoCTF-shell/hacksport/deploy.py

```diff
@@ -41,10 +41,15 @@ def give_port():
     """
 
     global port_random
-
     context = get_deploy_context()
-
+    
+    used_ports = []
```

Comment:
Can we keep the used_port list persistent so it doesn't have to rebuild per deployment?